packager/media/chunking/chunking_handler.cc

```diff
@@ -31,6 +31,7 @@ bool IsNewSegmentIndex(int64_t new_index, int64_t current_index) {
 ChunkingHandler::ChunkingHandler(const ChunkingParams& chunking_params)
     : chunking_params_(chunking_params) {
   CHECK_NE(chunking_params.segment_duration_in_seconds, 0u);
+  segment_number_ = chunking_params.start_segment_number - 1;
```

Comment:
Why is it `start_segment_number - 1`?

It does not seem to be necessary.

Right now `GetSegmentName` is based on `segment_index`. It should be modified to be based on `segment_number`.

Reply:
`Why is it start_segment_number - 1?`
I was treating segment_number as segment_index. Will change it.